src/cargo/ops/registry.rs

```diff
@@ -34,6 +34,11 @@ pub fn publish(manifest_path: &Path,
                verify: bool) -> CargoResult<()> {
     let pkg = try!(Package::for_path(&manifest_path, config));
 
+    if *pkg.private() {
+        bail!("private crates cannot be published to crates.io.\n\
+        `{}` is a private crate", pkg.name());
```

Comment:
Are we considering internal `crates.io` repositories? If so, maybe we should say `"private crates cannot be published to this repository..."` instead of explicitly naming `crates.io`.